crates/burn-import/src/burn/node/layer_norm.rs

```diff
@@ -0,0 +1,177 @@
+#[derive(Debug, Clone)]
+pub struct LayerNormNode<PS: PrecisionSettings> {
+    pub field: OtherType,
+    pub input: TensorType,
+    pub output: TensorType,
+    pub gamma: DataSerialize<PS::FloatElem>,        // Scale
+    pub beta: Option<DataSerialize<PS::FloatElem>>, // Bias (B)
+    pub config: LayerNormConfig,
+    pub full_precision: bool,
+}
+
+impl<PS: PrecisionSettings> LayerNormNode<PS> {
+    pub fn new<S: AsRef<str>>(
+        name: S,
+        input: TensorType,
+        output: TensorType,
+        gamma: DataSerialize<PS::FloatElem>,
+        beta: Option<DataSerialize<PS::FloatElem>>,
+        config: LayerNormConfig,
+        full_precision: bool,
+    ) -> Self {
+        Self {
+            field: OtherType::new(
+                name,
+                quote! {
+                    LayerNorm<B>
+                },
+            ),
+            input,
+            output,
+            gamma,
+            beta,
+            config,
+            full_precision,
+        }
+    }
+}
+
+impl<PS: PrecisionSettings> NodeCodegen<PS> for LayerNormNode<PS> {
+    fn input_types(&self) -> Vec<Type> {
+        vec![Type::Tensor(self.input.clone())]
+    }
+    fn output_types(&self) -> Vec<Type> {
+        vec![Type::Tensor(self.output.clone())]
+    }
+    fn field_type(&self) -> Option<Type> {
+        Some(Type::Other(self.field.clone()))
+    }
+
+    fn field_init(&self) -> Option<TokenStream> {
+        let name = &self.field.name;
+        let num_features = self.config.d_model.to_tokens();
+        let epsilon = self.config.epsilon;
+
+        let tokens = quote! {
+            let #name = LayerNormConfig::new(#num_features)
+                .with_epsilon(#epsilon)
+                .init(device);
+        };
+
+        Some(tokens)
+    }
+
+    fn field_serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        let device = Default::default();
+        let record = LayerNormRecord::<SerializationBackend> {
+            gamma: Param::initialized(
+                ParamId::new(),
+                Tensor::from_data(self.gamma.clone().convert(), &device),
+            ),
+            beta: Param::initialized(
+                ParamId::new(),
+                if let Some(beta) = self.beta.clone() {
+                    Tensor::from_data(beta.convert(), &device)
+                } else {
+                    Tensor::zeros([self.config.d_model], &device)
+                },
+            ),
+            epsilon: ConstantRecord::new(),
+        };
+
+        let item = Record::into_item::<PS>(record);
+        item.serialize(serializer)
+    }
+
+    fn forward(&self, scope: &mut Scope, node_position: usize) -> TokenStream {
+        let input = scope.tensor_use_owned(&self.input, node_position);
+        let output = &self.output.name;
+        let field = &self.field.name;
+
+        // TODO: handle self.full_precision
+        quote! {
+            let #output = self.#field.forward(#input);
+        }
```

Comment:
Was thinking we could match the elem precision and call `into_full_precision()` but the backendbridge requires the backend to be specified and here this is a generic... so can't do that right now.